common/src/main/java/net/william278/husktowns/hook/Pl3xMapHook.java

```diff
@@ -0,0 +1,147 @@
+package net.william278.husktowns.hook;
+
+import net.pl3x.map.core.Pl3xMap;
+import net.pl3x.map.core.event.EventHandler;
+import net.pl3x.map.core.event.EventListener;
+import net.pl3x.map.core.event.server.Pl3xMapEnabledEvent;
+import net.pl3x.map.core.event.world.WorldLoadedEvent;
+import net.pl3x.map.core.event.world.WorldUnloadedEvent;
+import net.pl3x.map.core.markers.Point;
+import net.pl3x.map.core.markers.layer.SimpleLayer;
+import net.pl3x.map.core.markers.marker.Marker;
+import net.pl3x.map.core.markers.option.Options;
+import net.pl3x.map.core.markers.option.Tooltip;
+import net.william278.husktowns.HuskTowns;
+import net.william278.husktowns.claim.TownClaim;
+import net.william278.husktowns.claim.World;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.Collection;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.logging.Level;
+import java.util.stream.Collectors;
+
+public class Pl3xMapHook extends MapHook implements EventListener {
```

Comment:
Okay so did some testing -- turns out Implementing `EventListener` here will cause the exception you described since the signature of the class depends on `net.pl3x.map.core.event.EventListener` being present on the classpath. Since we import Pl3xMapHook to reference it in the main class, we'll need to make a static inner class that holds the event listening logic :)

Reply:
This should be fixed as of [f7f1d3c](https://github.com/WiIIiam278/HuskTowns/pull/303/commits/f7f1d3c64ad4142d417727ef7f1264578d9c838e)